lib/cryptoexchange/exchanges/basefex/services/contract_stat.rb

```diff
@@ -0,0 +1,51 @@
+module Cryptoexchange::Exchanges
+  module Basefex
+    module Services
+      class ContractStat < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            false
+          end
+        end
+
+        def fetch
+          output = super ticker_url
+          adapt_all(output)
+        end
+
+        def ticker_url
+          "#{Cryptoexchange::Exchanges::Basefex::Market::API_URL}/instruments"
+        end
+
+        def adapt_all(output)
+          output.map do |pair|
+            base, target = Basefex::Market.separate_symbol(pair["symbol"])
+            market_pair = Cryptoexchange::Models::MarketPair.new(
+                            base: base,
+                            target: target,
+                            inst_id: pair["symbol"],
+                            contract_interval: "perpetual",
+                            market: Basefex::Market::NAME
+                          )
+
+            adapt(pair, market_pair)
+          end
+        end
+
+        def adapt(output, market_pair)
```

Comment:
seems like the api return index price

set (try verify with the contract spec doc on the index price to make sure it's correct or close)

`contract_stat.open_interest = output['indexPrice']`

Reply:
https://www.basefex.com/docs/spec-BTCUSD